File: codewars/6kyu/Rectangle into Squares/main.go
package kata

func SquaresInRect(lng int, wdth int) []int {
	if lng == wdth {
		return nil
	}

	var res []int
	length, width := lng, wdth

	for length > 0 && width > 0 {
		if length > width {
			res = append(res, width)
			length -= width
		} else {
			res = append(res, length)
			width -= length
		}
	}

	return res
}

File: codewars/6kyu/Rectangle into Squares/main.rs
fn sq_in_rect(lng: i32, wdth: i32) -> Option<Vec<i32>> {
    if lng == wdth {
        return None;
    }

    let mut res = Vec::new();

    let mut length = lng;
    let mut width = wdth;

    while length > 0 && width > 0 {
        if length > width {
            res.push(width);
            length -= width;
        } else {
            res.push(length);
            width -= length;
        }
    }

    Some(res)
}
